# Time Value of Money And Interest Rate: Finance Fundementals.

In this repository , we will master topics like time value of money concepts: Compounding , discounting , present value, annnuities, and interest rate conventions essentials for quantitative finance.

## 1. Time Value of Money and Interest Rates

A dollar today is worth more that dollar tomorrow. The simple obeservation underpins virutally every financial calculation you will encounter,from pricing bonds and valuing companies to structuring mortgages and retirment plans. The concept is called the **Time Value of Money**(TVM) and understanding it is essential for anyone working in quantitative finance.

Why is money worth more today? Three resasons driven this phenonmenon.

1. Money received today can be invested to ear a return , making it grow over time.

2. Inflation erodes purchasing power so a dollar today buys more than a dollar in the future.

3. There is uncertainity about whether promised future payements will actually materalize. These factors combine to create a fundamentals asymmetry: Present money is more valuable that future money.

The Time value of money gives us a framework for comparing each flows that occur at different points in time. Without this framework, how would you choose between receiving $1,000 today versus $1,100 in one year? Or evaluate a business that requires $1 million upfront but promises $200,000 annually for the next seven years? TVM provides the mathematical machinery to answer these questions rigorously.


Interest rate are the mechanism that links present and future values. They quantify the "price" of borrowing more or the "reward" for lending it. In qunatitative finance, interest rates serve as the foundation for pricing nearly every finance instrument. Understanding how they work , how they compound and how they are quoted across markets is ciritical knowledge.


## 2. Future Value and Compounding

When you deposit money in an interest-bearing account, your balance grows over time. The process by which this occurs and paticularly how interest accumulates on previously earned interest is called compounding.This concept drives both wealth accumulation and debt growth. Let's build up from the simplest case to the continuous limit used in derivatives pricing, understanding each step along the way.

## 4. Simple Interest:

Simple interest is the most basic form of interest calculation and is our conceptual starting point. Under simple interest, the interest accrues only on the original principal amount, never on accumulated interest. Think of it as a linear process: each period adds the same fixed amount of interest, regardless of how much has accumulated before. If you invest a principal 
**P** at an annual interest rate **r** for **t** years, the future value is:

$$ FV = P(1 + r * t)$$

where:
- **FV**: future value of the investment
- **P**: principal amount (initial investment)
- **r**: annual interest rate (expressed as a decimal)
- **t**: time in years

The structure of this formula reveals its linear nature. The term 
**rt** represents the total proportional increase: the rate 
**r** multiplied by time 
**t**. When you add this to 1 (representing the original principal as 100%), you get the total growth factor. Multiplying by 
**P**
**P** then scales this to your actual investment amount. Notice that time and rate enter symmetrically and additively. Doubling the time has exactly the same effect as doubling the rate.

> Simple interest is calculated only on the original principal amount. The interest earned each period remains constant regardless of how long the investment is held.


For example, investing `$1,000` at `5%` simple interest for `3` years yields:



In [1]:
# Simple Interest Calculation
P = 1000
r = 0.05 
t = 3

FV_Simple = P *( 1 + r * t)
print(f"The Future Value :{FV_Simple}")
interest_earned = FV_Simple - P
print(f"Interest earned on the Principal {P} :{interest_earned}")

The Future Value :1150.0
Interest earned on the Principal 1000 :150.0


Lets create a simple Interest function so we can reuse it again:

In [1]:

def SimpleInterest(P,R,T)-> int:
    SimpleInterest = P * ( 1 + R * T)
    return SimpleInterest

Notice that the interest earned ($150) equals the annual interest ($50) multiplied by the number of years (3). Simple interest grows linearly with time, which makes it straightforward to calculate but unrealistic for most financial applications. In the real world, lenders and investors expect to earn interest on their accumulated interest, a concept we turn to next.

## 5. Discrete Comopounding

In Practise, interest typically compounds periodically. This means the interest earned in each perdio is added to the principal and subsequent interest is calculated on this larget balance. The Fundamental insight is that your money earns returns on your original investment and on the returns that investment has already generated.This created exponential rather than linear growth a distinction with profound implications over long time horizons.

To understna why compounding leads to expoential growth, consider what happens period by period. After the first compounding period, your palance is $P(1+r/n)$. After the seond period, interest is calculated on this new, larger balance , giving $P(1+r/n)(1+r/n) = P(1+r/n)^{2}$. Each perdio multiplies the previous balance by the same growth factor, and multiplication repeated many times produces exponential behaviour.

For investment compounding $n$ times per year at annual rate $r$, the future value after $t$ years is :

$$ FV = P ( 1 + r / n)^{nt}$$


where:
- **FV** : future value after 
- **t** :  years
- **P** : principal amount invested
- **r** :  nominal annual interest rate (stated rate)
- **n** : number of compounding periods per year
- **t**: time in years
- $(1+r/n)$: growth factor per compounding period
- $nt$: total number of compounding periods

